app/pages/dashboard/getQueries.js

```diff
@@ -0,0 +1,160 @@
+import _ from 'lodash';
+import { utils as vizUtils } from '@tidepool/viz';
+import utils from '../../core/utils';
+
+const chartPrefs = {
+  basics: {
+    stats: {
+      excludeDaysWithoutBolus: false,
+    },
+    sections: {},
+    extentSize: 14,
+  },
+  daily: {
+    extentSize: 1,
+  },
+  trends: {
+    activeDays: {
+      monday: true,
+      tuesday: true,
+      wednesday: true,
+      thursday: true,
+      friday: true,
+      saturday: true,
+      sunday: true,
+    },
+    activeDomain: '2 weeks',
+    extentSize: 14,
+    // we track both showingCbg & showingSmbg as separate Booleans for now
+    // in case we decide to layer BGM & CGM data, as has been discussed/prototyped
+    showingCbg: true,
+    showingSmbg: false,
+    smbgGrouped: false,
+    smbgLines: false,
+    smbgRangeOverlay: true,
+
+    // Formerly in viz.trends redux store
+    cbgFlags: {
+      cbg100Enabled: true,
+      cbg80Enabled: true,
+      cbg50Enabled: true,
+      cbgMedianEnabled: true,
+    },
+    focusedCbgDateTrace: null,
+    focusedCbgSlice: null,
+    focusedCbgSliceKeys: null,
+    focusedSmbg: null,
+    focusedSmbgRangeAvg: null,
+    showingCbgDateTraces: false,
+    touched: false,
+    stats: {
+      excludeDaysWithoutBolus: false,
+    },
+  },
+  bgLog: {
+    bgSource: 'smbg',
+    extentSize: 14,
+  },
+  agpBGM: {
+    bgSource: 'smbg',
+  },
+  agpCGM: {
+    bgSource: 'cbg',
+  },
+  settings: {
+    touched: false,
+  },
+  excludedDevices: [],
+};
+
+const getQueries = (
+  data,
+  patient,
+  clinic,
+  opts,
+) => {
+  const bgPrefs = (() => {
+    const patientSettings = patient?.settings || {};
+
+    const bgUnitsOverride = {
+      units: clinic?.preferredBgUnits,
+      source: 'preferred clinic units',
+    };
+
+    const localBgPrefs = utils.getBGPrefsForDataProcessing(patientSettings, bgUnitsOverride);
+    localBgPrefs.bgBounds = vizUtils.bg.reshapeBgClassesToBgBounds(localBgPrefs);
+
+    return localBgPrefs;
+  })();
+
+  const timePrefs = (() => {
+    const latestTimeZone = data?.metaData?.latestTimeZone;
+    const queryParams = {};
+
+    const localTimePrefs = utils.getTimePrefsForDataProcessing(latestTimeZone, queryParams);
+    
+    return localTimePrefs;
+  })();
```

Comment:
Working backwards in the callback chain to get to the `queries` arg, I get to these 2 values which are defined by `this.state.bgPrefs` and `this.state.timePrefs`. Because they are held in component state, the process of how they get their value is a little nebulous.

From code reading and testing, it seems like `bgPrefs` represent the `clinic's` preferences for bg readings, while timePrefs represent the _patient's_ device preferences for time. Is this understanding correct?

I _believe_ they are defined [here in componentWillReceiveProps](https://github.com/tidepool-org/blip/blob/WEB-3336-tide-drawer/app/pages/patientdata/patientdata.js#L1694-L1718). Unfortunately, I can't simply copy and paste that code because it is coupled to component lifecycle, so I have written this batch of code to get us `bgPrefs` and `timePrefs`, and I think it works and is accurate. 

Would be awesome if you could help me double check this work.